core/unix/os.c

```diff
@@ -7501,6 +7582,19 @@ pre_system_call(dcontext_t *dcontext)
         dcontext->sys_param1 = (reg_t)func_arg;
         *sys_param_addr(dcontext, 0) = (reg_t)new_bsdthread_intercept;
         *sys_param_addr(dcontext, 1) = (reg_t)clone_rec;
+
+#    ifdef X64
+        /* Also update the pthread->fun and pthread->arg fields, since _pthread_start uses
+         * them instead of the syscall arg0 on some macOS versions */
+        ASSERT(sys_param(dcontext, 3) &&
```

Comment:
I don't think we should ever assert on an app-supplied value: we want to be able to run any app.  If the parameter is null, just skip this code gracefully.  Use ASSERT_CURIOSITY if you want a non-fatal notification, or SYSLOG_INTERNAL_WARNING for a less visible one.